src/ray/raylet/node_manager.cc

```diff
@@ -261,6 +261,35 @@ void NodeManager::ProcessNewClient(std::shared_ptr<LocalClientConnection> client
   client->ProcessMessages();
 }
 
+void NodeManager::DispatchTasks() {
+  // Work with a copy of scheduled tasks.
+  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  // Return if there are no tasks to schedule.
+  if (scheduled_tasks.empty()) {
+    return;
+  }
+  // Early return if there are no resources available.
+  const ClientID &my_client_id = gcs_client_->client_table().GetLocalClientId();
```

Comment:
Does it make sense to move this inside the for loop, in case the resources go to zero after assignment of a task?